File: HtProject-sp-callback/src/com/database/LightModel/tbl_cityItem.java
package com.database.LightModel;

import java.util.*;

import com.database.Logical.LightDataModel;

import java.math.BigDecimal;

public class tbl_cityItem extends LightDataModel {

	public final static String identifyField = "id";
	public final static String tableName = "tbl_city";

	/** 主键值 */
	private int _id;

	/** 省份ID */
	private int _province_id;
	/** 城市名称 */
	private String _name;

	@Override
	public String IdentifyField() {
		return identifyField;
	}

	@Override
	public String TableName() {
		return tableName;
	}

	@Override
	protected String[] GetNullableFields() {
		return new String[] { null, "province_id", "name" };
	}

	public Boolean Is_province_idNull() {
		return IsNull(Fields.province_id);
	}

	public void Set_province_idNull() {
		SetNull(Fields.province_id);
	}

	public Boolean Is_nameNull() {
		return IsNull(Fields.name);
	}

	public void Set_nameNull() {
		SetNull(Fields.name);
	}

	public int get_id() {
		return this._id;
	}

	public void set_id(int value) {
		this._id = value;
	}

	/** 省份ID */
	public int get_province_id() {
		return this._province_id;
	}

	/** 省份ID */
	@SuppressWarnings("unused")
	public void set_province_id(int value) {
		RemoveNullFlag(Fields.province_id);
		SetFieldHasUpdate(Fields.province_id, this._province_id, value);
		this._province_id = value;
	}

	/** 城市名称 */
	public String get_name() {
		return this._name;
	}

	/** 城市名称 */
	@SuppressWarnings("unused")
	public void set_name(String value) {
		if (false && true)
			RemoveNullFlag(Fields.name);
		else if (!false) {
			if (value == null)
				SetNullFlag(Fields.name);
			else
				RemoveNullFlag(Fields.name);
		}

		SetFieldHasUpdate(Fields.name, this._name, value);
		this._name = value;
	}

	/** 数据表字段列表对像 */
	public final class Fields {
		private Fields() {
		}

		public final static String id = "id";
		/** 主键 */
		public final static String PrimaryKey = "id";

		/** 省份ID */
		public final static String province_id = "province_id";
		/** 城市名称 */
		public final static String name = "name";

	}

}

File: xtone-dotnet-interface/n8wan.public/LightDataModel/tbl_cityItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace LightDataModel
{
    /// <summary>
    /// tbl_city数据模型
    /// </summary>
    public partial class tbl_cityItem : Shotgun.Model.Logical.LightDataModel
    {
        /// <summary>
        /// 数据表字段列表对像
        /// </summary>
        public sealed class Fields
        {
            private Fields() { }
            #region 表字段名称

            public const string id = "id";
            ///<summary>
            ///主键
            ///</summary>
            public const string PrimaryKey = "id";

            /// <summary>
            /// 省份ID
            /// </summary>
            public const string province_id = "province_id";
            /// <summary>
            /// 城市名称
            /// </summary>
            public const string name = "name";

            #endregion

        }
        #region 表字段变量定义
        /// <summary>
        /// 省份ID
        /// </summary>
        private int _province_id;
        /// <summary>
        /// 城市名称
        /// </summary>
        private string _name;

        #endregion

        public override string IdentifyField { get { return identifyField; } }
        public override string TableName { get { return tableName; } }

        public static readonly string identifyField = "id";
        public static readonly string tableName = "tbl_city";


        public int id
        {
            get;
            set;
        }

        #region 表字段值存取
        /// <summary>
        /// 省份ID
        /// </summary>
        public int province_id
        {
            get { return this._province_id; }
            set
            {
#if true && true
                RemoveNullFlag(Fields.province_id);
#elif !true
			    if (value == null)
                    SetNullFlag(Fields.province_id);
                else
                    RemoveNullFlag(Fields.province_id);
#endif

                SetFieldHasUpdate(Fields.province_id, this._province_id, value);
                this._province_id = value;
            }
        }
        /// <summary>
        /// 城市名称
        /// </summary>
        public string name
        {
            get { return this._name; }
            set
            {
#if false && true
				RemoveNullFlag(Fields.name);
#elif !false
                if (value == null)
                    SetNullFlag(Fields.name);
                else
                    RemoveNullFlag(Fields.name);
#endif

                SetFieldHasUpdate(Fields.name, this._name, value);
                this._name = value;
            }
        }

        #endregion
        #region 空值相关方法
        protected override string[] GetNullableFields()
        {
            return new string[]{null
			,"province_id"
,"name"
};
        }
        public bool Isprovince_idNull() { return IsNull(Fields.province_id); }

        public void Setprovince_idNull() { SetNull(Fields.province_id); }
        public bool IsnameNull() { return IsNull(Fields.name); }

        public void SetnameNull() { SetNull(Fields.name); }

        #endregion
        #region 静态方法
        /// <summary>
        /// 创建一个新的查询器
        /// </summary>
        public static Shotgun.Model.List.LightDataQueries<tbl_cityItem> GetQueries(Shotgun.Database.IBaseDataClass2 dBase)
        {
            return new Shotgun.Model.List.LightDataQueries<tbl_cityItem>(tableName, identifyField, dBase);
        }

        /// <summary>
        /// 创建一个新的查询器
        /// </summary>
        public static Shotgun.Model.List.LightDataQueries<tbl_cityItem> GetQueries()
        {
            return new Shotgun.Model.List.LightDataQueries<tbl_cityItem>(tableName, identifyField);
        }


        /// <summary>
        /// 根据主键查找指定的行,返回所有字段
        /// </summary>
        /// <param name="dBase"></param>
        /// <param name="id">主键值</param>
        /// <returns></returns>
        public static tbl_cityItem GetRowById(Shotgun.Database.IBaseDataClass2 dBase, int id)
        {
            return GetRowById(dBase, id, null);
        }


        #endregion
    }
}